src/app/containers/AssignmentDutyAddModal.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { default as AssignmentDutyCreateForm } from '../containers/AssignmentDutyCreateForm';
+import {
+    Button,
+    Glyphicon
+} from 'react-bootstrap';
+import ModalWrapper from './ModalWrapper';
+import { IdType } from '../api';
+
+export interface AssignmentDutyAddModalProps {
+    assignmentId: IdType;
+}
+export default class AssignmentDutyAddModal extends React.Component<AssignmentDutyAddModalProps> {
+    render() {
+        const { assignmentId } = this.props;
+        return (
+            <ModalWrapper
+                title="Add a Duty"
+                showButton={({ handleShow }) =>
```

Comment:
I've been placing parenthesis around blocks of jsx.  It allows for multiline jsx and also will be auto formatted how you'd expect.  Here you would have

```ts
showButton={({ handleShow }) => (
                <Button bsStyle="primary" bsSize="xsmall" onClick={() => handleShow()}>
                     <Glyphicon glyph="plus" />
                </Button>
            )}
```